boa3_test/examples/HTLC.py

```diff
@@ -0,0 +1,234 @@
+
+
+@public
+def atomic_swap(owner_address: UInt160, owner_token: bytes, owner_amount: int, other_person_address: UInt160,
+                other_person_token: bytes, other_person_amount: int, secret_hash: bytes) -> bool:
+    """
+    Initializes the storage when the atomic swap starts.
+
+    :param owner_address: address of owner
+    :type owner_address: UInt160
+    :param owner_token: other_person's desired token
+    :type owner_token: bytes
+    :param owner_amount: other_person's desired amount of tokens
+    :type owner_amount: int
+    :param other_person_address: address of other_person
+    :type other_person_address: bytes
+    :param other_person_token: owner's desired token
+    :type other_person_token: bytes
+    :param other_person_amount: owner's desired amount of tokens
+    :type other_person_amount: int
+    :param secret_hash: the secret hash created by the contract deployer
+    :type secret_hash: bytes
+
+    :return: whether the deploy was successful or not
+    :rtype: bool
+
+    :raise AssertionError: raised if `owner_address` or `other_person_address` length is not 20 or if `amount` is not
+    greater than zero.
+    """
+    # the parameters from and to should be 20-byte addresses. If not, this method should throw an exception.
+    assert len(owner_address) == 20 and len(other_person_address) == 20
+    # the parameter amount must be greater than 0. If not, this method should throw an exception.
+    assert owner_amount > 0 and other_person_amount > 0
+
+    if get(NOT_INITIALIZED).to_bool() == True and verify():
+        put(ADDRESS_PREFIX + OWNER, owner_address)
+        put(TOKEN_PREFIX + OWNER, owner_token)
+        put(AMOUNT_PREFIX + OWNER, owner_amount)
+        put(ADDRESS_PREFIX + OTHER_PERSON, other_person_address)
+        put(TOKEN_PREFIX + OTHER_PERSON, other_person_token)
+        put(AMOUNT_PREFIX + OTHER_PERSON, other_person_amount)
+        put(SECRET_HASH, secret_hash)
+        put(NOT_INITIALIZED, False)
+        put(START_TIME, get_time)
+        return True
+    return False
+
+
+@public
+def onPayment(from_address: UInt160, amount: int, data: Any):
+    """
+    Since this is a deployed contract, transfer will be calling this onPayment method with the data parameter from
+    transfer. If someone is doing a not required transfer, then ABORT will be called.
+
+    :param from_address: the address of the one who is trying to transfer cryptocurrency to this smart contract
+    :type from_address: UInt160
+    :param amount: the amount of cryptocurrency that is being sent to this smart contract
+    :type amount: int
+    :param data: any pertinent data that may validate the transaction
+    :type data: Any
+
+    :raise AssertionError: raised if `from_address` length is not 20
+    """
+    # the parameters from and to should be 20-byte addresses. If not, this method should throw an exception.
+    assert len(from_address) == 20
+
+    if get(NOT_INITIALIZED).to_bool() == False:
```

Comment:
again, are there any problems when you use `not get(NOT_INITIALIZED).to_bool()`?